packages/cardano-services/test/Asset/AssetHttpService.test.ts

```diff
@@ -0,0 +1,125 @@
+import { fromSerializableObject } from '@cardano-sdk/util';
+import { getPort } from 'get-port-please';
+import { dummyLogger as logger } from 'ts-log';
+import axios from 'axios';
+
+const APPLICATION_JSON = 'application/json';
+const APPLICATION_CBOR = 'application/cbor';
+const UNSUPPORTED_MEDIA_STRING = 'Request failed with status code 415';
+const BAD_REQUEST_STRING = 'Request failed with status code 400';
+
+describe('AssetHttpService', () => {
+  let apiUrlBase: string;
+  let config: HttpServerConfig;
+  let db: Pool;
+  let httpServer: HttpServer;
+  let service: AssetHttpService;
+  let port: number;
+
+  beforeAll(async () => {
+    port = await getPort();
+    apiUrlBase = `http://localhost:${port}/asset`;
+    config = { listen: { port } };
+  });
+
+  describe('healthy state', () => {
+    beforeAll(async () => {
+      db = new Pool({ connectionString: process.env.DB_CONNECTION_STRING });
+      service = new AssetHttpService({ db, logger });
+      httpServer = new HttpServer(config, { services: [service] });
+      await httpServer.initialize();
+      await httpServer.start();
+    });
+
+    afterAll(async () => {
+      await httpServer.shutdown();
+      await db.end();
+    });
+
+    describe('/health', () => {
+      it('/health response should be true', async () => {
+        const res = await axios.post(`${apiUrlBase}/health`, undefined, {
+          headers: { 'Content-Type': APPLICATION_JSON }
+        });
+        expect(res.status).toBe(200);
+        expect(res.data).toEqual({ ok: true });
+      });
+    });
+
+    describe('/get-asset', () => {
+      it('returns a 415 coded response if the wrong content type header is used', async () => {
+        try {
+          await axios.post(`${apiUrlBase}/get-asset`, { args: [] }, { headers: { 'Content-Type': APPLICATION_CBOR } });
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+          expect(error.response.status).toBe(415);
+          expect(error.message).toBe(UNSUPPORTED_MEDIA_STRING);
+        }
+      });
+
+      it('returns 400 coded response if the request is bad formed', async () => {
+        try {
+          await axios.post(`${apiUrlBase}/get-asset`, { args: [['test']] });
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+          expect(error.response.status).toBe(400);
+          expect(error.message).toBe(BAD_REQUEST_STRING);
+        }
+      });
+
+      it('internal server error for not existing existing asset id', async () => {
+        try {
+          const res = await axios.post(`${apiUrlBase}/get-asset`, {
+            args: ['0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef']
+          });
+          expect(res.data[0]).toEqual({});
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+          expect(error.response.status).toBe(500);
```

Comment:
```suggestion
          expect(error.response.status).toBe(404);
```

Reply:
Sorry @rhyslbw and @mkazlauskas ,

I was trying to propagate `ProviderError{ProviderFailure.NotFound}` through the HTTP response as we said [here](https://github.com/input-output-hk/cardano-js-sdk/pull/321#discussion_r924423559); is there already some helper in the sdk to do that?

I mean the [router](https://github.com/input-output-hk/cardano-js-sdk/blob/feat/ADP-1606-asset-http-service/packages/cardano-services/src/Asset/AssetHttpService.ts#L57-L60) is using sdk standard; the [getAssist](https://github.com/input-output-hk/cardano-js-sdk/blob/feat/ADP-1606-asset-http-service/packages/cardano-services/test/Asset/DbSyncAssetProvider.test.ts#L40-L42) method already throw the right error...

If I have to write some custom code, could you please point me a code sample to see how to propagate a `ProviderError` through the HTTP response?

Thank you

Reply:
[This](https://github.com/input-output-hk/cardano-js-sdk/blob/cd7458e7c2550f5c7a6d98b3dfacb8d0ea5ce2ea/packages/cardano-services/src/Http/HttpService.ts#L59) catch should work

Reply:
Hi @rhyslbw and @mkazlauskas ,

it seems the given `catch` didn't exactly fit as it was always responding with http status code `500`; anyway f872fec9f introduces the small change to let it work as required.